docs/architecture/state.md

```diff
@@ -1,91 +1,111 @@
-* Parallel transactions executed concurrently by distinct actors.
-* Concurrent state model that allows block production without knowing the full state.
+Miden’s state model focuses on:
 
-Privacy is enforced by a UTXO-like state model consisting of notes and nullifiers combined with off-chain execution using zero-knowledge proofs.
+- **Privacy:**  
+  By using notes and nullifiers, Miden ensures that value transfers remain confidential. Zero-knowledge proofs allow users to prove correctness without revealing sensitive information.
 
-State bloat describes the ever growing state stored in blockchain nodes. Polygon Miden addresses this challenge via its state model that enables concurrent off-chain execution and off-chain storage. Simply put, users can store their own data locally which reduces the burden on the network while integrity is ensured using zero-knowledge.
+- **Flexible data storage:**  
+  Users can store data privately on their own devices or within the network. This approach reduces reliance on the network for data availability, helps maintain user sovereignty, and minimizes unnecessary on-chain storage.
 
-## State components
+- **Concurrency:**  
+  Multiple transactions can be processed concurrently by distinct actors which improves throughput and efficiency.
 
-Miden nodes maintain three databases to describe state:
+Miden’s state model supports a private, secure, and high-throughput environment while also addressing the challenges of state bloat commonly associated with continuously growing blockchain states.
 
-1. A database of accounts.
-2. A database of notes.
-3. A database of nullifiers for already consumed notes.
+## What is state?
 
-![Architecture core concepts](../img/architecture/state/state.png)
+The state of the Miden rollup describes the current condition of all accounts and notes in the protocol; i.e., the current reality.
+
+## State model components
 
-These databases are represented by authenticated data structures that enable easy proof of items added to or removed from a database. This ensures that the commitment to the database remains very small.
+The Miden node maintains three databases to describe state:
 
-Polygon Miden has two databases to capture the note states. The note database is append-only and stores all notes permanently. The nullifier database stores nullifiers that indicate that a note has been previously consumed. Separating note storage into these two databases gives Polygon Miden client-side proving and advanced privacy.
+1. Accounts
+2. Notes
+3. Nullifiers
+
+![Architecture core concepts](../img/architecture/state/state.png)
 
 ### Account database
 
-The latest account states - and data for on-chain accounts - are recorded in a sparse Merkle tree which maps account IDs to account hashes, and account data if needed.
+The accounts database stores the latest account states for public accounts or state commitments for private accounts.
```

Comment:
I think it may be good to add some more detail here. For example, account DB has 2 purposes: (1) track state commitments of all accounts in an authenticated data structure (i.e., the SMT), (2) store all account data for public accounts. This would also make the graphic below a bit easier to understand.